google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITFloat32Test.java

```diff
@@ -0,0 +1,448 @@
+  }
+
+  private Struct readLastRow(String... columns) {
+    return readRow("T", lastKey, columns);
+  }
+
+  @Test
+  public void writeFloat32() {
+    assumeTrue("FLOAT32 is currently only supported in cloud-devel", isUsingCloudDevel());
+
+    write(baseInsert().set("Float32Value").to(2.0f).build());
+    Struct row = readLastRow("Float32Value");
+    assertThat(row.isNull(0)).isFalse();
+    assertThat(row.getFloat(0)).isWithin(0.0f).of(2.0f);
+  }
+
+  @Ignore("Needs a fix in the backend.")
+  @Test
+  public void writeFloat32NonNumbers() {
+    assumeTrue("FLOAT32 is currently only supported in cloud-devel", isUsingCloudDevel());
+
+    write(baseInsert().set("Float32Value").to(Float.NEGATIVE_INFINITY).build());
+    Struct row = readLastRow("Float32Value");
+    assertThat(row.isNull(0)).isFalse();
+    assertThat(row.getFloat(0)).isNegativeInfinity();
+
+    write(baseInsert().set("Float32Value").to(Float.POSITIVE_INFINITY).build());
+    row = readLastRow("Float32Value");
+    assertThat(row.isNull(0)).isFalse();
+    assertThat(row.getFloat(0)).isPositiveInfinity();
+
+    write(baseInsert().set("Float32Value").to(Float.NaN).build());
+    row = readLastRow("Float32Value");
+    assertThat(row.isNull(0)).isFalse();
+    assertThat(row.getFloat(0)).isNaN();
+  }
+
+  @Test
+  public void writeFloat32Null() {
+    assumeTrue("FLOAT32 is currently only supported in cloud-devel", isUsingCloudDevel());
+
+    write(baseInsert().set("Float32Value").to((Float) null).build());
+    Struct row = readLastRow("Float32Value");
+    assertThat(row.isNull(0)).isTrue();
+  }
+
+  @Test
+  public void writeFloat32ArrayNull() {
+    assumeTrue("FLOAT32 is currently only supported in cloud-devel", isUsingCloudDevel());
+
+    write(baseInsert().set("Float32ArrayValue").toFloat32Array((float[]) null).build());
+    Struct row = readLastRow("Float32ArrayValue");
+    assertThat(row.isNull(0)).isTrue();
+  }
+
+  @Test
+  public void writeFloat32ArrayEmpty() {
+    assumeTrue("FLOAT32 is currently only supported in cloud-devel", isUsingCloudDevel());
+
+    write(baseInsert().set("Float32ArrayValue").toFloat32Array(new float[] {}).build());
+    Struct row = readLastRow("Float32ArrayValue");
+    assertThat(row.isNull(0)).isFalse();
+    assertThat(row.getFloatList(0)).containsExactly();
+  }
+
+  @Test
+  public void writeFloat32Array() {
+    assumeTrue("FLOAT32 is currently only supported in cloud-devel", isUsingCloudDevel());
+
+    write(
+        baseInsert()
+            .set("Float32ArrayValue")
+            .toFloat32Array(Arrays.asList(null, 1.0f, 2.0f))
+            .build());
+    Struct row = readLastRow("Float32ArrayValue");
+    assertThat(row.isNull(0)).isFalse();
+    assertThat(row.getFloatList(0)).containsExactly(null, 1.0f, 2.0f).inOrder();
+    try {
+      row.getFloatArray(0);
+      fail("Expected exception");
+    } catch (NullPointerException ex) {
+      assertNotNull(ex.getMessage());
+    }
```

Comment:
Use `assertThrows(...)` instead of this construct. This type of construct was needed back when we had to support Java 7.

Reply:
Ack. Apologies, this was a copy paste from an older test.